lib/data/external/GCP/GcpUtils.js

```diff
@@ -1 +1,30 @@
-module.exports = {};
+const werelogs = require('werelogs');
+
+const _config = require('../../../Config.js').config;
+
+werelogs.configure({
+    level: _config.log.logLevel,
+    dump: _config.log.dumpLevel,
+});
+
+const logger = new werelogs.Logger('gcpUtil');
+
+class JsonError extends Error {
+    constructor(type, code, desc) {
+        super(type);
+        this.code = code;
+        this.description = desc;
+        this[type] = true;
+    }
+}
+
+function checkSourceString(sourceString) {
+    return Boolean(sourceString.match(/^\/{0,1}[^/]+(\/[^/]+)+$/g));
```

Comment:
I would have to check, but it seems like a source object with multiple slashes (ex. `a/b/c`) should not be valid. Do you know? 

In any case, instead of the regexp I think perhaps it would be simpler to validate using `.split('/')`, especially since you call that method in a subsequent step.

Reply:
Multiple slashes should be a valid source string. Example, in `a/b/c`, `a` is the bucket and `b/c` is the object key.

Reply:
True. The matching should also factor in source object with trailing slashes then (ex. `a/b/c/`).